private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -711,3 +698,362 @@ a co-author.
+with the EKT Key.  All endpoints in the conference utilize
+the same SRTP master salt value for a given EKT Key.
+
+The EKT Field in media packets is encrypted using a cipher specified
+via the EKTKey message (e.g., AES Key Wrap).  This cipher is different
+than the cipher used to protect media and is only used to encrypt
+the endpoint's SRTP master key (and other EKT Field data as per
+[@!I-D.ietf-perc-srtp-ekt-diet]).
+
+The media distributor is not given the KEK (i.e., EKT Key).
+
+## Endpoints fabricate an SRTP Master Key
+
+As stated earlier, the E2E key received via DTLS-SRTP is discarded.
+While it could have been used, the fact that endpoints may need to
+change the SRTP master key periodically means using it has only limited
+utility.  To reduce complexity, PERC prescribes that endpoints
+manufacturer random SRTP master keys locally to be used for E2E
+encryption.
+
+Since the media server is not involved in E2E functions, it will not
+create this key.  Note, too, that even the key distributor is unaware
+of the locally-generated E2E keys used by each endpoint.
+
+## Who has What Key
+
+To summarize the various keys and which entity is in possession
+of a given key, refer to (@fig-who-has-what-key).
+
+All endpoints have knowledge of the KEK and it is the same for
+all participants.
+
+Every HBH key is distinct at a given endpoint, thus Endpoint A and
+endpoint B do not have knowledge of the other's HBH key.
+
+Each endpoint generates its own E2E Key (SRTP master key), thus the
+key distinct per endpoint.  This key is transmitted (encrypted) via
+the EKT Field to other endpoints.  Endpoints that receive media from
+a given transmitting endpoint will have knowledge of the transmitter's
+E2E key.
+
+{#fig-who-has-what-key align="center"}
+~~~
++----------------------+------------+-------+-------+------------+
+| Key     /    Entity  | Endpoint A |  MD X |  MD Y | Endpoint B |
++----------------------+------------+-------+-------+------------+
+| KEK                  |    Yes     |  No   |  No   |     Yes    |
++----------------------+------------+-------+-------+------------+
+| E2E Key (A and B)    |    Yes     |  No   |  No   |     Yes    |
++----------------------+------------+-------+-------+------------+
+| HBH Key (A<=>MD X)   |    Yes     |  Yes  |  No   |     No     |
++----------------------+------------+-------+-------+------------+
+| HBH Key (B<=>MD Y)   |    No      |  No   |  Yes  |     Yes    |
++----------------------+------------+---------------+------------+
+| HBH Key (MD X<=>MD Y)|    No      |  Yes  |  Yes  |     No     |
++----------------------+------------+---------------+------------+
+~~~
+Figure: Keys per Entity
+
+# PERC Packet Format {#packetformat}
+
+This appendix is intended to aid the reader in understanding the PERC media
+packet format by walking through the steps required to construct the packet
+at each step.
+
+## PERC Packet Formation Steps
+
```

Comment:
As a meta comment on this whole section of "PERC Packet Formation Steps", I think how the packet is formed is largely overlapping and perhaps not 100% in sync with double and it is going to be hard to get to agreement on what it says. However, it seems much easier to get to agreement on what does the packet on the wire look like. My proposal would be mostly remove this section and next but keep the "PERC Packet Format" section and get that to be much more explicit of what happens over wire for a specific example.